### PR TITLE
Firearm Report Realism

### PR DESCRIPTION
**Pull request by maxmurder**
This PR adds a base loudness to firearms and tweaks suppressors for added realism. This makes guns MUCH louder (base loudness for most guns is 150), prolonged fire will cause deafness unless the player is wearing hearing protection. Suppressors will bring most guns' reports down to generally safe levels.

Most guns follow this pattern for base loudness:
Standard guns (most rifles and pistols): 150
Revolvers: 160
Shotguns: 175
Sawn-off/Launchers/Big guns: 200
Pneumatic/Spring: 5
Energy and rail weapons have no base loudness. 

[CR] Because this has major balance considerations I would like as much feedback as possible.

Enhancements: Add muzzle flash effect, possibly add chaff with chance to start fires. Will work on these features in a separate PR though.

### CODE REVIEW

**Comment by Rivet-the-Zombie** [comment]
>prolonged fire will cause deafness unless the player is wearing hearing protection

How much are we talking here? I know from personal experience that I can fire my .357 magnum and 12 gauge shotgun all day (100+ rounds through the .357 and 200+ through the shotgun in one day) without hearing protection and the worst I have to worry about is a bit of ringing that doesn't last more than an hour or so after I stop.

I really like this idea, I just don't want firearms to end up being silly about making noise to the point where they're unusable.

**Comment by John-Candlebury** [comment]
Also, you may want  update the gun onomatopeyas, so some guns still go "bang", instead of all going "Kerblam"

**Comment by maxmurder** [comment]
@rivet: ATM the effect is fairly noticeable, firing a burst of 8 ak-47 rounds will leave you deaf for a few turns, In single fire mode it is more manageable but you still have a chance of getting deafness. 150 + ammo damage puts most guns just over the chance for deafness threshold, so you only have a small chance of getting deafness for each shot. Firing individual shots should only deafen a player for a very short time, whereas firing rapidly or while already deafened has a chance to add to the duration of deafness. As it is now even prolonged firing will only result in very temporary deafness.
From what I understand deafness mechanic is meant to simulate the shell-shock/ear ringing immediately after a loud noise. Simulating long-term permanent damage from firing guns all day without hearing protection would an interesting addition but as a separate feature.

@John: This change pushes most guns well into "Kerblam" territory.

**Comment by i2amroy** [comment]
I think what he's getting at is that you might want to broaden the size of the gun onomatopoeia so bang would be like 10-40 instead of 10-20 (just pulling these numbers out of the air, but that's the idea).

**Comment by maxmurder** [comment]
Ah, reading comprehension fail.

**Comment by kevingranade** [comment]
This is going to have a side effect of guns always pulling every enemy in the reality bubble to your position, and hordes will probably move toward you from entire overmaps away.

**Comment by Rivet-the-Zombie** [comment]
That's really how it ought to be. On open terrain I can tell where gunshots are coming from for *miles*.

**Comment by i2amroy** [comment]
Personally if we had to adjust for that I'd vote at turning down the propensity of a horde to track you down from a single gunshot rather then reducing their volume. There's most likely a variety of other things that could distract a horde's attention at distances like that, and it seems kinda silly that a horde will come for you from forever and a half away from a single gunshot or two, even if they heard them clearly.

**Comment by KA101** [comment]
>horde gunfire-homing stuff

This is starting to sound like a reality-bubble issue.  By rights, if the horde is near a miloutpost or other turret-defended area, the turrets should start firing on the horde, and the horde should deal with those much closer shots rather than your small-arms hunting.

Somehow I suspect we're not likely to account for that.

**Comment by maxmurder** [comment]
Hmm, these are some good points. I had neglected to think about horde-tracking. I usually play with dynamic spawning/wander spawns off. After a little testing with those enabled, this change doesn’t make dynamic spawning that much more intense than before, the biggest difference is that now all guns cause spawns and not just shotguns etc. Hordes seem track you from a few tiles away at least, I need to dig into how horde sound tracking works. I wanted more individual zombies in your reality bubble getting pulled, as in the current system most guns are only slightly louder (and many are quieter) than a breaking window or an engine idling.

**Comment by kevingranade** [comment]
Hmm, we're drifting quite a bit off-topic, but we certainly can account for that sort of thing by representing turrets and zombies as hordes on the overmap, and letting them interact there in a simplified way.  You could potentially lead a zombie horde to an outpost or roadblock, and then let the resulting conflict pull in zombies from a large area.  Unless something particular happened though, I would expect the random background noises to cancel out, and you end up with the net effect on hordes to be negligible unless a particular sound source persists over a long time or is very close to the horde.  This would be the justification for reducing how much interest a horde takes in a particular sound source.
I don't see any of this as a blocker for this PR though, hordes are experimental and broken, and I don't mind breaking them a bit more right now, this shouldn't make them any harder to fix in the future.

**Comment by maxmurder** [comment]
Ok, good to hear. How do you guys feel about the current volume levels? should they be reduced?

**Comment by maxmurder** [comment]
OK i am leaving the values as-is for now. It will be easy enough to tweak in the future if people feel deafness caused by guns is too OP.

File: data/json/items/mods.json [hunk 1]
```diff
@@ -87,7 +87,7 @@
         "name" : "homemade suppressor",
         "dispersion_modifier" : 15,
         "damage_modifier" : -5,
-        "description" : "Using a suppressor is almost an imperative in zombie-infested regions.  Gunfire is very noisy, and will attract predators.  Its only drawback is a reduced muzzle velocity, resulting in less accuracy and damage.",
+        "description" : "A suppressor reduces the amount of noise and muzzle flash generated by the report of a firearm. Gunfire is extremely loud and can damage your hearing without protection, a suppressor will bring the loudness of a report down to generally safe levels. A suppressor will slightly reduce recoil and muzzle velocity.",
```
Comment (i2amroy) [inline data/json/items/mods.json]: Small thing, but you should be using two spaces after each period on both this and the normal suppression description.

**Pull request merged**